cpp/include/cudf/functions.h

```diff
@@ -912,3 +906,45 @@ gdf_error gdf_quantile_aprrox(	gdf_column*  col_in,       //input column with 0
                                 double       q,            //requested quantile in [0,1]
                                 void*        t_erased_res, //type-erased result of same type as column;
                                 gdf_context* ctxt);        //context info
+
+
+/* --------------------------------------------------------------------------*/
+/** 
+ * @brief Sorts an array of gdf_column in ascending order.
+ * 
+ * @Param[in] input_columns Array of gdf_columns
+ * @Param[in] num_inputs # columns 
+ * @Param[in] flag_nulls_are_smallest Flag to indicate if nulls are to be considered
+ *                                    smaller than non-nulls or viceversa
+ * @Param[out] output_indices Pre-allocated gdf_column to be filled with sorted
+ *                            indices
+ * 
+ * @Returns GDF_SUCCESS upon successful completion
+ */
+/* ----------------------------------------------------------------------------*/
+gdf_error gdf_order_by(gdf_column** input_columns,
+                       size_t       num_inputs,
+                       gdf_column*  output_indices,
+                       int          flag_nulls_are_smallest);
+
+/* --------------------------------------------------------------------------*/
+/** 
+ * @brief Sorts an array of gdf_column given the sort order for each column.
+ * 
+ * @Param[in] input_columns Array of gdf_columns
+ * @Param[in] asc_desc Device array of sort order types for each column
+ *                     (0 is ascending order and 1 is descending)
+ * @Param[in] num_inputs # columns
+ * @Param[in] flag_nulls_are_smallest Flag to indicate if nulls are to be considered
+ *                                    smaller than non-nulls or viceversa
+ * @Param[out] output_indices Pre-allocated gdf_column to be filled with sorted
+ *                            indices
+ * 
+ * @Returns GDF_SUCCESS upon successful completion
+ */
+/* ----------------------------------------------------------------------------*/
+gdf_error gdf_order_by_asc_desc(gdf_column** input_columns,
```

Comment:
The more I look at this API, the more I think we should just consolidate these two functions into a single function `gdf_order_by` that accepts an optional `asc_desc` argument. If `asc_desc == nullptr`, then just assume all ascending.

Reply:
Hello Jake, I appreciate the feedback, but this change like any changes the the public API can cause other disruptions. We are currently using this API, so we would really rather not change it

Reply:
A decision to change the API should not be based on how it's currently used. It should be based on impacted LOCs, design, and ease-of-use.

Maintaining two APIs for order-by means that we have to build two sets of python bindings, and two sets of docs for the two functions.

Given that all that really needs to be done is add a `nullptr` parameter to the order-by calls, I think it would be appropriate to change the API here to save us work/headache in the future.